### PR TITLE
mpd: Support blacklisting MPD commands in the server.

### PR DESCRIPTION
**Pull request by adamcik**
Default blacklist set to listall and listallinfo.

This change has been done to avoid clients being able to call "bad" MPD
commands which are often misused to try and keep a client db. Note that
this change will break some MPD clients, but the blacklist can be controlled
via config to allow opting out for now.

This is relevant as a first step in the direction suggested in https://github.com/mopidy/mopidy/issues/686#issuecomment-64089734 which advocates getting rid of the commands all together.

### CODE REVIEW

**Comment by trygveaa** [comment]
It is worth noting that these two commands (listall and listallinfo) has been deprecated by MPD:

"Do not use this command. Do not manage a client-side copy of MPD's database. That is fragile and adds huge overhead. It will break with large databases. Instead, query MPD whenever you need something."

http://www.musicpd.org/doc/protocol/database.html

File: docs/changelog.rst [hunk 1]
```diff
@@ -84,6 +84,11 @@ v0.20.0 (UNRELEASED)
 - Share a single mapping between names and URIs across all MPD sessions. (Fixes:
   :issue:`934`, PR: :issue:`968`)
 
+- Add support for blacklisting MPD commands. This is used to prevent clients
+  from using `listall` and `listallinfo` which recursively lookup the entire
```
Comment (jodal) [inline docs/changelog.rst]: Double quotes. This isn't Markdown :-)

File: docs/ext/mpd.rst [hunk 2]
```diff
@@ -99,3 +99,10 @@ See :ref:`config` for general help on configuring Mopidy.
     ``$hostname`` and ``$port`` can be used in the name.
 
     Set to an empty string to disable Zeroconf for MPD.
+
+.. confval:: mpd/command_blacklist
+
+    List of MPD commands which are disabled by the server. By default this
+    setting blacklists `listall` and `listallinfo`. These commands don't fit
```
Comment (jodal) [inline docs/ext/mpd.rst]: Double quotes again.

File: mopidy/mpd/exceptions.py [hunk 3]
```diff
@@ -87,3 +87,12 @@ class MpdNotImplemented(MpdAckError):
     def __init__(self, *args, **kwargs):
         super(MpdNotImplemented, self).__init__(*args, **kwargs)
         self.message = 'Not implemented'
+
+
+class MpdDisabled(MpdAckError):
```
Comment (jodal) [inline mopidy/mpd/exceptions.py]: Add a comment stating that this error is specific to Mopidy?

File: mopidy/utils/deprecation.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
 
 
 def _is_pykka_proxy_creation():
+    return False
```
Comment (jodal) [inline mopidy/utils/deprecation.py]: I'm guessing this wasn't supposed to be added, as @trygveaa pointed out.

**Comment by jodal** [comment]
+1 with comments

**Comment by adamcik** [comment]
Done.

File: mopidy/mpd/exceptions.py [hunk 5]
```diff
@@ -87,3 +87,13 @@ class MpdNotImplemented(MpdAckError):
...
+    # NOTE: this is a custom error for mopidy that does not exists in MPD.
```
Comment (jodal) [inline mopidy/mpd/exceptions.py]: s/this/This/
s/mopidy/Mopidy/
s/exists/exist/

**Comment by adamcik** [comment]
...and fixed.

**Pull request merged**